python/ray/ml/trainer.py

```diff
@@ -210,8 +241,74 @@ def fit(self) -> Result:
             TrainingFailedError: If any failures during the execution of
             ``self.as_trainable()``.
         """
-        raise NotImplementedError
+        trainable = self.as_trainable()
+
+        from ray import tune
+        from ray.tune import TuneError
+
+        # TODO(amog/xwjiang): Replace with Tuner.
+        try:
+            analysis = tune.run(run_or_experiment=trainable, **self.run_config)
```

Comment:
Is this tested? Doesn't seem to work.

Reply:
Yeah this is being tested in `test_trainer`. What is not working?

Reply:
run_config is RunConfig now. Also test_trainer is not passing in any run_config.

Reply:
Ok removed, we will update later.